charts/k8ssandra-cluster/templates/cassandra/cassdc.yaml

```diff
@@ -34,6 +32,11 @@ spec:
       resources:
         requests:
           storage: {{ .Values.k8ssandra.cassandraLibDirVolume.size | default "5Gi" }}
+{{- if .Values.k8ssandra.configuration.auth.superuser.secret }}
+  superuserSecretName: {{ .Values.k8ssandra.configuration.auth.superuser.secret }}
+{{- else if .Values.k8ssandra.configuration.auth.superuser.username }}
```

Comment:
The commit says `where a secret will be generated with random password`, but I don't see that behavior in this commit? Also, what happens if the secret does not exists which the user defines? What happens to the deployment?

Reply:
It's there now. My initial commit left it out.

>what happens if the secret does not exists which the user defines?

I believe the CassandraDatacenter deployment will fail. I need to verify this though.

Reply:
If the secret does not exist, here is what cass-operator logs:

`{"level":"error","ts":1610336606.837807,"logger":"reconciliation_handler","msg":"CassandraDatacenter resource is invalid","requestNamespace":"default","requestName":"dc1","loopID":"d4014dd7-b580-4368-ba54-37a2dbe82906","error":"Could not load superuser secret for CassandraCluster: default/missing-secret","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/datastax/cass-operator/operator/pkg/reconciliation.(*ReconcileCassandraDatacenter).Reconcile\n\t/cass-operator/operator/pkg/reconciliation/handler.go:239\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:88"}`

`{"level":"info","ts":1610336606.8378613,"logger":"reconciliation_handler","msg":"Could not load superuser secret for CassandraCluster: default/missing-secret","requestNamespace":"default","requestName":"dc1","loopID":"d4014dd7-b580-4368-ba54-37a2dbe82906","reason":"ValidationFailed","eventType":"Warning"}
{"level":"info","ts":1610336606.8379087,"logger":"reconciliation_handler","msg":"Reconcile loop completed","requestNamespace":"default","requestName":"dc1","loopID":"d4014dd7-b580-4368-ba54-37a2dbe82906","duration":0.00029109}`

`{"level":"error","ts":1610336606.83793,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"cassandradatacenter-controller","request":"default/dc1","error":"Could not load superuser secret for CassandraCluster: default/missing-secret","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:88"}`

Nothing is reported in the CassandraDatacenter status to indicate that there is a problem. I am going to create a cass-operator issue for this.

Reply:
See https://github.com/datastax/cass-operator/issues/352